code/modules/mob/dead/new_player/login.dm

```diff
@@ -3,8 +3,12 @@
 		return
 
 	if(CONFIG_GET(flag/use_exp_tracking))
-		client.set_exp_from_db()
-		client.set_db_player_flags()
+		client?.set_exp_from_db()
+		client?.set_db_player_flags()
```

Comment:
both of these should have been moved to client/New() instead. on-connection events should not be happening in Login()

Reply:
I can move them aye